lib/Doctrine/ORM/Cache/DefaultCollectionHydrator.php

```diff
@@ -81,15 +81,34 @@ public function loadCacheEntry(ClassMetadata $metadata, CollectionCacheKey $key,
         $targetRegion    = $targetPersister->getCacheRegion();
         $list            = array();
 
+        $keys = array();
         foreach ($entry->identifiers as $index => $identifier) {
+            $keys[$index] = new EntityCacheKey($assoc['targetEntity'], $identifier);
+        }
+
 
-            $entityEntry = $targetRegion->get(new EntityCacheKey($assoc['targetEntity'], $identifier));
+        if ($targetRegion instanceof MultiGetRegion) {
```

Comment:
Yes, but changing an interface would be a BC break.... (is it an internal BC break?)